src/main/java/com/avispl/symphony/dal/device/core510i/QSYSCoreCommunicator.java

```diff
@@ -336,4 +430,312 @@ private void populateQSYSMonitoringMetrics(Map<String, String> stats) {
 			throw new ResourceNotReachableException(errBuilder.toString());
 		}
 	}
+
+	/**
+	 * This method is used for populate Q-SYS gain controlling metrics
+	 *
+	 * @param stats is the map that store all statistics
+	 * @param controllableProperties is the list that store all controllable properties
+	 */
+	private void populateGainControllingMetrics(Map<String, String> stats, List<AdvancedControllableProperty> controllableProperties) {
+		// If user doesn't input anything, just break out the method
+		if (gain == null) {
+			return;
+		}
+
+		String[] namedGainComponents = handleGainInputFromUser();
+
+		GetComponentsResponse namedComponents = (GetComponentsResponse) getResponsesFromDevice(new GetComponentsRequest(), GetComponentsResponse.class);
+
+		// If there is no component in the design file -> just return and do nothing
+		if (namedComponents == null) {
+			return;
+		}
+
+		for (String namedGainComponent : namedGainComponents) {
+			GainControlInfo controlInfo;
+			String errorMessage;
+
+			try {
+				controlInfo = getControlInfoFromComponent(namedComponents, namedGainComponent);
+				// replace # by ~ due to Symphony doesn't accept #
+				namedGainComponent = namedGainComponent.replace(QSYSCoreConstant.HASH, QSYSCoreConstant.TIDE);
+
+				if (controlInfo == null) {
+					errorMessage = QSYSCoreConstant.GETTING_DEVICE_INFO_ERR;
+
+					if (this.logger.isDebugEnabled()) {
+						this.logger.debug(errorMessage);
+					}
+
+					populateUnavailableGainControllingGroup(stats, namedGainComponent, errorMessage);
+				} else {
+					double minGain = Double.parseDouble(controlInfo.getMinGain());
```

Comment:
controlInfo.getMinGain() and controlInfo.getMaxGain() can be null or NumberFormatException . Please handle it!

Reply:
Thank you, I fixed it